ament_cmake_core/cmake/index/ament_index_register_package.cmake

```diff
@@ -29,5 +29,5 @@ function(ament_index_register_package)
   endif()
 
   # register package name with the resource type 'packages'
-  ament_index_register_resource("packages" ${ARGN})
+  ament_index_register_resource("packages" ${ARG_PACKAGE_NAME})
```

Comment:
Why did you change this? This is not how the API is supposed to be used. See https://github.com/ament/ament_cmake/blob/ebeb5994632c3d1e9f116a6869fea7a80a5a6af5/ament_cmake_core/cmake/index/ament_index_register_resource.cmake#L19-L33